### ISSUE TITLE
Suboptimal recommendation to use elvis operator 

### ISSUE THREAD

**Issue by stevenh** [nvuillam/npm-groovy-lint#297]
The follow code triggers a warning for the use of an elvis operator to assign a default value:
```groovy
  if (!platforms[platform]) {
      platforms[platform] = new Platform(name: platform, id: platformID, subSystems: [])
  }
```
```log
Code could use elvis operator: platforms[platform] = platforms[platform] ?: new Platform([name:platform, id:platformID, subSystems:[]])   CouldBeElvis
```

This would introduce an unnecessary assign to the map value, so it doesn't seem like the best advise.

Should the warning trigger in this case?

**Comment by stevenh** [comment]
Now I know more of how this is put together I'm guessing this is a CodeNarc issue?

**Comment by nvuillam** [comment]
@stevenh if this is a groovy error, it is raised by CodeNarc yes :)
And the CodeNarc used within npm-groovy-lint is not up to date :(
